utilities/message_filters/src/message_filters/__init__.py

```diff
@@ -184,9 +185,11 @@ def __init__(self, fs, queue_size):
 
     def connectInput(self, fs):
         self.queues = [{} for f in fs]
-        self.input_connections = [f.registerCallback(self.add, q) for (f, q) in zip(fs, self.queues)]
+        self.input_connections = [
+            f.registerCallback(self.add, q, i_q)
+            for i_q, (f, q) in enumerate(zip(fs, self.queues))]
 
-    def add(self, msg, my_queue):
+    def add(self, msg, my_queue, my_queue_index):
```

Comment:
This changes existing public API and would therefore break existing code. I don't think this is feasible for an already released ROS distro.